src/inmanta/loader.py

```diff
@@ -230,6 +237,11 @@ def install_source(self, module_source: ModuleSource) -> bool:
                 os.path.join(all_modules_dir, pathlib.PurePath(pathlib.PurePath(relative_module_path).parts[0]))
             )
 
+            if module_source.is_bytecode():
+                init_file = "__init__.pyc"
+            else:
+                init_file = "__init__.py"
+
```

Comment:
I'm a bit concerned about this. What if a module contains only one byte compiled file. e.g.
```
inmanta_plugins/mymod/__init__.py  # contains some plugins
inmanta_plugins/mymod/resources.pyc
```
If `install_source` is called in this order, then first `__init__.py` will be written, followed by `resources.pyc`. The loading of the latter will create `__init__.pyc`, which will be an empty file that gets precendence over the actual `__init__.py` file.

Reply:
We could update this logic but we might just as well consider not doing anything if it's a byte-compiled file, because `touch_inits` is only relevant for orchestrator versions of the modules v1 era anyway. At least, that's how it's documented.

Reply:
A second remark here is that I don't think an empty `pyc` file is allowed. I get
```
Traceback (most recent call last):
  File "/home/sander/downloads/bytecompiletest/main.py", line 1, in <module>
    import mymod
ImportError: bad magic number in 'mymod': b''
```

So if we would keep this `touch_inits`, we probably need to byte-compile the empty string first.

Reply:
Fixed now